namespaces/resource-namespace.md

```diff
@@ -0,0 +1,231 @@
+# Resource Namespace
+
+This document extends [Refract][] [MSON Namespace][] to define REST Resource data structure elements.
+
+## About this Document
+
+This document conforms to RFC 2119, which says:
+
+> The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+[MSON][] is used throughout this document.
+
+## Supporting Data Types
+
+### Href (string)
+
+The value of `Href` is to be used as a URI Template, as defined in RFC 6570.
+The value of the `Href` is a template used to determine the target URI of the related resource.
+The value of the instance property SHOULD be resolved as a URI-Reference per RFC 3986 and MAY be a relative reference to a URI.
+
+## Namespace Elements
+
+### Href Variables (Object Type)
+
+The definition is a MSON namespace `Object Type` where keys are respective URI Template variables.
+
+#### Properties
+
+- `element`: `href-variables` (string, fixed)
+
+### Data Structure (MSON Element)
+
+Data structure definition using MSON namespace elements.
+
+#### Properties
+
+- `element`: `data-structure` (string, fixed)
+
+### Resource (Element)
+
+The Resource representation with its available transitions and attributes.
+
+#### Properties
+
+- `element`: resource (string, fixed)
+- `attributes` (object)
+    - `href` (Href) - URI Template of this resource.
+    - `hrefVariables` (Href Variables) - Definition of URI Template variables used in the `href` property.
+- `content` (array)
+    - (Transition) - State transition available for this resource.
+
+        Note: The `content` array MAY include multiple `Transitions`.
+
+    - (Data Structure) - Data structure associated with the resource.
+
+        Note: The `content` array MUST include exactly one `Data Structure` at maximum.
+
+#### Example
+
+```json
+{
+    "element": "resource",
+    "meta": {
+        "title": "Question",
+        "description": "A Question object has the following attributes."
+    },
+    "attributes": {
+        "href": "/questions/{question_id}",
+        "hrefVariables": {
+            "element": "href-variables",
```

Comment:
@danielgtaylor Definitely want to work on clarity here, but what it was supposed to mean is that you should rely on the definition of an element to know what has been refracted versus what may be refracted. 

In other words, if the element you are using defines an attribute as having an element as its value, when you refract the object, you know what has already been refracted. Without relying on that definition, you can't really know.

Hope that makes sense!